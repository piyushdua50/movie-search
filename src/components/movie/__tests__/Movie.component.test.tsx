import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import Movie from "../Movie.component";
test("render the Movie correctly", () => {
  const mockStore = configureStore();
  const mockedMovie = {
    Title: "Kuch Tum Kaho Kuch Hum Kahein",
    Year: "2002",
    imdbID: "tt0327004",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BNGVhMjBhYmQtMDFmZC00Mjc2LWIwZTYtMmJiY2ZkNjM4N2FlXkEyXkFqcGdeQXVyNDUzOTQ5MjY@._V1_SX300.jpg",
  };
  const { container } = render(
    <BrowserRouter>
      <Provider store={mockStore()}>
        <Routes>
          <Route path="/" element={<Movie movie={mockedMovie} />} />
        </Routes>
      </Provider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
