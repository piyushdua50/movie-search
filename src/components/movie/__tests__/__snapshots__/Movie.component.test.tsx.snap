// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the Header correctly 1`] = `
<div>
  <div
    class="movieContainer"
  >
    <img
      alt="Kuch Tum Kaho Kuch Hum Kahein"
      class="imgContainer"
      src="https://m.media-amazon.com/images/M/MV5BNGVhMjBhYmQtMDFmZC00Mjc2LWIwZTYtMmJiY2ZkNjM4N2FlXkEyXkFqcGdeQXVyNDUzOTQ5MjY@._V1_SX300.jpg"
    />
    <span
      class="movieNameContainer"
    >
      Kuch Tum Kaho Kuch Hum Kahein
    </span>
    <div
      class="movieInfoContainer"
    >
      <span
        class="movieInfo"
      >
        Year
        : 
        2002
      </span>
      <span
        class="movieInfo"
      >
        Type
        : 
        movie
      </span>
    </div>
    <div
      class="movieFavoriteInfo"
      role="button"
    >
      Favorite
    </div>
  </div>
</div>
`;

exports[`render the Movie correctly 1`] = `
<div>
  <div
    class="movieContainer"
  >
    <img
      alt="Kuch Tum Kaho Kuch Hum Kahein"
      class="imgContainer"
      src="https://m.media-amazon.com/images/M/MV5BNGVhMjBhYmQtMDFmZC00Mjc2LWIwZTYtMmJiY2ZkNjM4N2FlXkEyXkFqcGdeQXVyNDUzOTQ5MjY@._V1_SX300.jpg"
    />
    <span
      class="movieNameContainer"
    >
      Kuch Tum Kaho Kuch Hum Kahein
    </span>
    <div
      class="movieInfoContainer"
    >
      <span
        class="movieInfo"
      >
        Year
        : 
        2002
      </span>
      <span
        class="movieInfo"
      >
        Type
        : 
        movie
      </span>
    </div>
    <div
      class="movieFavoriteInfo"
      role="button"
    >
      Favorite
    </div>
  </div>
</div>
`;
